lib/validation.js

```diff
@@ -73,11 +73,34 @@ function validateParam (validatorFunction, request, paramName) {
 }
 
 function validate (context, request) {
-  return validateParam(context[paramsSchema], request, 'params') ||
-    validateParam(context[bodySchema], request, 'body') ||
-    validateParam(context[querystringSchema], request, 'query') ||
-    validateParam(context[headersSchema], request, 'headers') ||
-    true
+  var params = validateParam(context[paramsSchema], request, 'params')
+  if (params) {
+    return wrapValidationError(params, 'params')
+  }
+  var body = validateParam(context[bodySchema], request, 'body')
+  if (body) {
+    return wrapValidationError(body, 'body')
+  }
+  var query = validateParam(context[querystringSchema], request, 'query')
+  if (query) {
+    return wrapValidationError(query, 'querystring')
```

Comment:
I think it would be better to use the string `'query'` here since that's the property on `request` (`request.query`) and it's the name used in the schema (`{ schema:  { query: ... } }`).

Reply:
No, the name used in the schema is `querystring`

Reply:
Oh you're right. That's confusing that `schema.querystring` validates `request.query`.

Reply:
Yep, a bit would like to know the reasons for this? 😄